### ISSUE TITLE
Flaky spec: Booth Officer with multiple booth assignments today

### ISSUE THREAD

**Issue by bertocq** [AyuntamientoMadrid/consul#1202]
Warning!
===
This spec probably only fails between 00:00 and 02:00 am Madrid time

# What
Tests that fail randomly are called "flakies", this one seems to be one:

**Randomized seed:** 19087

**Failure:**

```
    1) Booth Officer with multiple booth assignments today
     Failure/Error: expect(page).to have_content 'Choose your booth'
       expected to find text "Choose your booth" in "Validate document Total recounts and results Language: English Español Français Nederlands Português Valencià How to help Menu Decide Madrid Polling Polling officers Notifications You don't have new notifications My activity My account Sign out Admin menu Validate document Total recounts and results × You don't have officing shifts today Poll officing Here you can validate user documents and store voting results"
     # ./spec/features/officing/booth_spec.rb:53:in `block (2 levels) in <top (required)>'
```

# How
- [ ] Explain why the test is flaky, or under which conditions/scenario it fails randomly
- [ ] Explain why your PR fixes it
- [ ] Create a backport PR to consul/consul when the fixing PR is approved

### Random values issues
If the problem comes from randomly generated values, running multiple times a single spec could help you reproduce the failure by running at your command line:
```bash
for run in {1..10}
do
  bin/rspec ./spec/features/budgets/investments_spec.rb:256
done
```

You can also try running a single spec in Travis:
Add option `:focus` to the spec and push your branch to Github, for example:
```ruby
scenario 'Show', :focus do
```

But remember to remove that `:focus` changes afterwards when submitting your PR changes!

### Test order issues
Running specs in the order they failed may discover that the problem is that a previous test sets an state in the test environment that makes our flaky fail/pass. Tests should be independent from the rest.

After executing rspec you can see the seed used, add it as an option to rspec, for example:
`bin/rspec --seed 55638` (check **Randomized seed** value at beginning of issue)

### Other things to watch for
- Time related issues (current time, two time or date comparisons with miliseconds/time when its not needed)
- https://semaphoreci.com/community/tutorials/how-to-deal-with-and-eliminate-flaky-tests

**Comment by bertocq** [comment]
happened at https://travis-ci.org/AyuntamientoMadrid/consul/jobs/338257557

**Comment by iagirre** [comment]
I'll do this.

**Comment by iagirre** [comment]
I have been studying this issue for a time now and I have a question about it. The thing is that the login of the officer is done through a function that uses the interface, instead of doing it using a `login_as officer` and redirecting to the target page after that, all in the same spec. Is any reason for doing it that way?

The function I'm talking about is: 
https://github.com/AyuntamientoMadrid/consul/blob/master/spec/support/common_actions.rb#L48

and it is used here:
https://github.com/AyuntamientoMadrid/consul/blob/master/spec/features/officing/booth_spec.rb#L51

**Comment by javierm** [comment]
Closed via consul#2712 and #1625.
